tick/optim/prox/prox_binarsity.py

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf8 -*-
+
+import numpy as np
+from .base import Prox
+from .build.prox import ProxBinarsity as _ProxBinarsity
+
+
+class ProxBinarsity(Prox):
+    """Proximal operator of binarsity, namely total-variation
+    regularization followed by centering within sub-blocks of the
+    coefficients. Blocks (non-overlapping) are specified by the
+    ``blocks_start`` and ``blocks_length`` parameters.
+
+    Parameters
+    ----------
+    strength : `float`, default=0.
+        Level of total-variation penalization
+
+    blocks_start : `np.array`, shape=(n_blocks,)
+        First entry of each block
+
+    blocks_length : `np.array`, shape=(n_blocks,)
+        Size of each block
+
+    range : `tuple` of two `int`, default=`None`
+        Range on which the prox is applied. If `None` then the prox is
+        applied on the whole vector
+
+    positive : `bool`, default=`False`
+        If True, apply L1 penalization together with a projection
+        onto the set of vectors with non-negative entries
+
+    Attributes
+    ----------
+    n_blocks : `int`
+        Number of blocks
+
+    Notes
+    -----
+    Uses the fast-TV algorithm described in:
+
+    * "A Direct Algorithm for 1D Total Variation Denoising"
```

Comment:
Original paper should be cited with a link (example in tick/inference/hawkes_em.py)